packages/frontend/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,24 @@
+import { Wrapper, Icon, StyledInput } from "./Search.styled";
+import loupe from "src/assets/images/loupe.svg";
+import { setFilterText } from "src/Views/Home/Pagination/reducer/actions/pagination.actions";
+import { usePaginationContext } from "src/Views/Home/Pagination/hooks/usePaginationContext";
+import debounce from "lodash.debounce";
+import { useRef } from "react";
+
+export const Search = () => {
+  const [, dispatch] = usePaginationContext();
+
+  const debounced = useRef(debounce((nextText) => setFilterText(dispatch, nextText), 1000)).current;
+
+  const handleChange = (e: React.FormEvent<HTMLInputElement>) => {
```

Comment:
handleChange powinien być przekazywany od komponentu wyżej. Żeby to był głupicomponent

Reply:
stan  wartości w inpucie powinien być trzymany w tym komoponecie na bierząco w useState, debounce najelpiej jakby był w useEffect i ustawiał interwał na 0.5 sekundy i dopiero po tym czasie ustawiał wartość w reducerze , nie ma potrzeby wyciągania stanu wyżej bo żaden inny komponent nie będzie z niego korzystał